integrations/huggingface/src/actions/generateContent.ts

```diff
@@ -0,0 +1,120 @@
+import * as sdk from "@botpress/sdk";
+import { interfaces } from "@botpress/sdk";
+import type { HfInference } from "@huggingface/inference";
+import { InvalidPayloadError } from "@botpress/client";
+
+type HfToolCalls = Awaited<
+  ReturnType<HfInference["chatCompletion"]>
+>["choices"][number]["message"]["tool_calls"];
+
+function mapToolCalls(
+  hfToolCalls: HfToolCalls
+): sdk.interfaces.llm.ToolCall[] | undefined {
+  return hfToolCalls
+    ?.filter((toolCall) => toolCall.type === "function")
+    .map((toolCall) => ({
+      id: `${toolCall.id}`,
+      type: toolCall.type as "function",
+      function: {
+        name: toolCall.function.name,
+        arguments: toolCall.function.arguments as Record<string, any> | null,
+      },
+    }));
+}
+
+function mapToStopReason(hfFinishReason: string) {
+  switch (hfFinishReason) {
+    case "length":
+      return "max_tokens";
+    case "stop":
+      return "stop";
+    case "content_filter":
+      return "content_filter";
+    case "tool_calls":
+      return "tool_calls";
+    default:
+      return "other";
+  }
+}
+
+export async function generateContent(
+  input: interfaces.llm.GenerateContentInput,
+  hf: HfInference,
+  models: interfaces.llm.Model[]
+): Promise<interfaces.llm.GenerateContentOutput> {
+  if (!input.model?.id) {
```

Comment:
this seems a bit odd to me. Why wouldn't the interface provide the model directly?

Reply:
I understand now!